src/main/java/nus/climods/logic/commands/DeleteCommand.java

```diff
@@ -23,19 +28,27 @@ public class DeleteCommand extends Command {
 
     /**
      * Creates a DeleteCommand with the given UserModule
-     * @param target UserModule to delete
+     * @param targetCode module code of UserModule to delete
      */
     public DeleteCommand(String targetCode) {
         requireNonNull(targetCode);
         this.targetCode = targetCode;
     }
 
+    private Module getModule(Model model, String moduleCode) throws CommandException {
+        Optional<Module> optionalModule = model.getModuleList().getListModule(moduleCode);
+
+        if (optionalModule.isEmpty()) {
+            throw new CommandException(MESSAGE_MODULE_NOT_FOUND);
+        }
+        return optionalModule.get();
+    }
+
     @Override
     public CommandResult execute(Model model) throws CommandException {
         requireNonNull(model);
 
-        UserModule toDelete = new UserModule(targetCode, model);
-
+        UserModule toDelete = new UserModule(getModule(model, targetCode));
```

Comment:
I am not sure if im understanding this wrong but for `delete` shouldn't be querying the user module list for existing user modules to delete? cmmiw but right now it seems like we doing a query on the main module list to create a new user module that is used to delete a user module based on equality (`equals`)

Reply:
I think switching to querying the user module list would make things must simpler? What do your think?

Reply:
i think filteredListhasUserModule is checking the filetredUserModuleList rather than the main list?

Reply:
yes but i dont see a reason why we need to create a new `UserModule` (`toDelete`) by querying the target module code it against the entire list of modules then pass it to check in `filteredListhasUserModule`

Reply:
Maybe we can have a method to query the just the target code against the `userModuleList` to check if it is in the list then delete it

Reply:
fixed!